common-custom-user-data-maven-extension/src/main/java/com/gradle/CommonCustomUserDataMavenExtension.java

```diff
@@ -25,13 +26,14 @@ public CommonCustomUserDataMavenExtension(PlexusContainer container, Logger logg
     @Override
     public void afterProjectsRead(MavenSession session) throws MavenExecutionException {
         logger.debug("Executing extension: " + getClass().getSimpleName());
-        BuildScanApi buildScan = BuildScanApiAccessor.lookup(container, getClass());
+        BuildScanApi buildScan = ApiAccessor.lookup(BuildScanApi.class, container, getClass());
         if (buildScan != null) {
             logger.debug("Capturing custom user data in build scan");
             CustomUserData.addToBuildScan(buildScan);
-            GroovyScriptUserData.addToBuildScan(session, buildScan, logger);
             logger.debug("Finished capturing custom user data in build scans");
         }
+        BuildCacheApi buildCache = ApiAccessor.lookup(BuildCacheApi.class, container, getClass());
+        GroovyScriptUserData.addToApis(session, buildScan, buildCache, logger);
```

Comment:
Let's do like for build scans and have a class called `CustomBuildCacheConfig` with `CustomBuildCacheConfig.configureBuildCache(BuildCacheApi buildCache). The implementation can be empty for now.

Reply:
Done